sparkplug/timereporters/base.py

```diff
@@ -0,0 +1,43 @@
+import datetime
+
+##################################################
+
+def _milliseconds(timedelta):
+    return timedelta / datetime.timedelta( microseconds=1000 )
+
+
+##################################################
+
+def mn_md_mx(x):
+    "calculate min median max from a list of timedeltas, returns the time in milliseconds"
+    mn = 0
+    median = 0
+    mx = 0
+    if x :
+        lenx = len(x)
+        ordered = sorted(x)
+        mn = _milliseconds(ordered[0])
+        mx = _milliseconds(ordered[-1])
+        if lenx & 0x01 :
```

Comment:
I wonder if using the `%` operator might make it more readable. Possibly something like
```lenx % 2 == 0```?